packages/web-components/src/components/va-table/va-table.tsx

```diff
@@ -41,110 +44,157 @@
-        'vads-u-background-color--gray-lightest',
-        'vads-u-color--base',
-      );
-      button.innerHTML = `${columns[this.sortColumn]}${icon}`;
-      button.onclick = this.handleSort.bind(this);
+    if (!this.uswds) {
+      // For IE11 compatibility. `el.children` renders booleans instead of html elements,
+      // so instead we use `el.childNodes` and filter out nodes that aren't a proper tag
+      const elementChildren = (el: HTMLElement) =>
+        Array.from(el.childNodes).filter(node => node.nodeName !== '#text');
 
-      const sortableHeader = this.headers[this.sortColumn];
-      sortableHeader.childNodes.forEach(child => child.remove());
-      sortableHeader.appendChild(button);
+      const headerRow = Array.from(
+        this.el.querySelectorAll('va-table-row'),
+      )[0] as HTMLVaTableRowElement;
 
-      this.handleSort();
-    }
+      this.headers = elementChildren(headerRow) as Array<HTMLElement>;
+      const columns = [];
 
-    // Store alignment classes by column index.
-    const alignment = {};
-
-    const handleCellAdjustments = () => {
-      const rows = Array.from(this.el.querySelectorAll('va-table-row'))
-        .filter((row) => row.slot !== 'headers') as Array<HTMLVaTableRowElement>;
-
-      Array.from(rows).forEach((row, index) => {
-        const cells = elementChildren(row);
-
-        cells.forEach((cell: HTMLSpanElement, colNum) => {
-          // Look at the first row of data to determine type of data in column
-          // Right align columns with numeric data
-          if (index === 0 && isNumeric(cell.textContent)) {
-            // eslint-disable-next-line i18next/no-literal-string
-            alignment[colNum] = 'text-align-right';
-            (this.headers[colNum] as HTMLElement).classList.add(
-              alignment[colNum],
-            );
-          }
-          if (alignment[colNum]) {
-            cell.classList.add(alignment[colNum]);
-          }
+      /* eslint-disable i18next/no-literal-string */
+      this.headers.forEach((item: HTMLVaTableRowElement) => {
+        columns.push(item.textContent);
+        item.setAttribute('role', 'columnheader');
+        item.setAttribute('scope', 'col');
+      });
+      /* eslint-enable i18next/no-literal-string */
+
+      if (this.sortColumn >= 0 && columns.length > this.sortColumn) {
+        const icon = this.sortAscending ? ascendingIcon : descendingIcon;
+        const button = document.createElement('button');
+        button.setAttribute(
+          'aria-label',
+          `sort data by ${this.sortAscending ? 'descending' : 'ascending'}`,
+        );
+        button.classList.add(
+          'vads-u-padding--0',
+          'vads-u-margin--0',
+          'vads-u-text-decoration--none',
+          'vads-u-font-weight--bold',
+          'vads-u-background-color--gray-lightest',
+          'vads-u-color--base',
+        );
+        button.innerHTML = `${columns[this.sortColumn]}${icon}`;
```

Comment:
Is it possible to add this button element to the rendered markup instead of dynamically? It would be nice to avoid having to use `innerHTML` if we could..

If not possible, then we should probably sanitize the content being passed into it using `Sanitizer.escapeHTML` or similar to address this warning.

USWDS provided us with some utility functions that we have adopted for va-file-input that might be useful:

https://github.com/department-of-veterans-affairs/component-library/blob/main/packages/web-components/src/utils/utils.ts#L128-L165

Reply:
@jamigibbs - the only affects the V1 component. It was flagged in this PR because I had to enclose the code in an `if` statement, so I think it is out of scope for this ticket?